x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/user_profile/40_has_privileges.yml

```diff
@@ -0,0 +1,176 @@
+        - stash_in_key
+
+  - do:
+      cluster.health:
+        wait_for_status: yellow
+
+  - do:
+      security.put_user:
+        username: "super"
+        body:  >
+          {
+            "password" : "s3krit-password",
+            "roles" : [ "superuser" ]
+          }
+
+  - do:
+      security.put_user:
+        username: "user1"
+        body:  >
+          {
+            "password": "s3krit-password",
+            "roles" : [ "role1" ]
+          }
+
+  - do:
+      security.put_role:
+        name: "role1"
+        body:  >
+          {
+            "indices": [
+              {
+                "names": "index1*",
+                "privileges": ["all"]
+              },
+              {
+                "names": "index2*",
+                "privileges": ["read"]
+              }
+            ]
+          }
+
+  - do:
+      security.put_user:
+        username: "user2"
+        body:  >
+          {
+            "password": "s3krit-password",
+            "roles" : [ "role2" ]
+          }
+
+  - do:
+      security.put_role:
+        name: "role2"
+        body:  >
+          {
+            "indices": [
+              {
+                "names": "index2*",
+                "privileges": ["all"]
+              },
+              {
+                "names": "index1*",
+                "privileges": ["read"]
+              }
+            ]
+          }
+
+---
+teardown:
+  - do:
+      security.delete_user:
+        username: "super"
+        ignore: 404
+  - do:
+      security.delete_user:
+        username: "user1"
+        ignore: 404
+  - do:
+      security.delete_user:
+        username: "user2"
+        ignore: 404
+  - do:
+      security.delete_role:
+        name: "role1"
+        ignore: 404
+  - do:
+      security.delete_role:
+        name: "role2"
+        ignore: 404
+
+---
+"Test profile has privileges api":
+
+  - do:
+      security.activate_user_profile:
+        body: >
+          {
+            "grant_type": "password",
+            "username": "user1",
+            "password" : "s3krit-password"
+          }
+  - match: { "user.username" : "user1" }
+  - is_true: uid
+  - set: { uid: profile_uid1 }
+
+  - do:
+      security.activate_user_profile:
+        body: >
+          {
+            "grant_type": "password",
+            "username": "user2",
+            "password" : "s3krit-password"
+          }
+  - match: { "user.username" : "user2" }
+  - is_true: uid
+  - set: { uid: profile_uid2 }
+
+  - do:
+      security.has_privileges_user_profile:
+        body:
+          uids:
+            - $profile_uid1
+            - $profile_uid2
+            - dummy_missing
+          privileges:
+            index:
+              - names:
+                  - index1*
+                  - index2*
+                privileges:
+                  - read
+  - length: { "has_privilege_uids": 2 }
+  - match: { "has_privilege_uids.0" : $profile_uid2 }
+  - match: { "has_privilege_uids.1" : $profile_uid1 }
```

Comment:
This is not stable since `has_privilege_uids` is a `Set`. You might have to use regex for it. It is not pretty but works. Here is an example:
https://github.com/elastic/elasticsearch/blob/23c41e45d9a20f43220d7a50d00dd373ab774394/x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/api_key/10_basic.yml#L273

Reply:
Thanks for the suggestion.

Upon further thought I think it would be reasonable to maintain the order from the request (the internal mget does preserve it). I vaguely remember considering it but I don't recall where we ended up with that. I think it would be easy to do.

Let me know what you think.